### PR TITLE
Adding nxos_vpc_interface

### PR DESCRIPTION
**Pull request by GGabriele**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

nxos_portchannel
##### ANSIBLE VERSION```

ansible 2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/']

```

##### SUMMARY
Manages interface VPC configuration

```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @GGabriele. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_vpc_interface.py [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+short_description: Manages interface VPC configuration
+description:
+    - Manages interface VPC configuration
+extends_documentation_fragment: nxos
+author:
+    - Jason Edelman (@jedelman8)
+    - Gabriele Gerbino (@GGabriele)
+notes:
+    - Either vpc or peer_link param is required, but not both.
+    - I(state)=absent removes whatever VPC config is on a port-channel
```
Comment (gundalow) [inline network/nxos/nxos_vpc_interface.py]: C(state=absent)

**Comment by gundalow** [comment]
Should @jedelman8 be listed as a maintainer of `network/nxos` along side you, so you both get notifications from Ansibull-bot. This will become more useful when other contributors start making changes to these modules

File: network/nxos/nxos_vpc_interface.py [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+short_description: Manages interface VPC configuration
+description:
+    - Manages interface VPC configuration
+extends_documentation_fragment: nxos
+author:
+    - Jason Edelman (@jedelman8)
+    - Gabriele Gerbino (@GGabriele)
+notes:
+    - Either vpc or peer_link param is required, but not both.
+    - C(state)=absent removes whatever VPC config is on a port-channel
```
Comment (gundalow) [inline network/nxos/nxos_vpc_interface.py]: `C(state=absent)`

**Comment by GGabriele** [comment]
@gundalow yes it will definitely help, thanks.

File: network/nxos/nxos_vpc_interface.py [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+    returned: always
+    type: boolean
+    sample: true
+'''
+
+
+import collections
+import json
+
+# COMMON CODE FOR MIGRATION
```
Comment (gundalow) [inline network/nxos/nxos_vpc_interface.py:99]: Is this going to be done for 2.2

**Comment by gundalow** [comment]
1) Can you please update the `descriptions:` for under `options:` to be full sentences (Capital letters & full stops). 

2) From your comments it looks like their is a load of common code that may need moving to `module_utils`. Do you have a plan for that?

**Comment by gundalow** [comment]
needs_revision

**Comment by gundalow** [comment]
Maintainers updated https://github.com/ansible/ansibullbot/commit/59fd66b10aba6c8027f232f845a4a33194b04d3f#diff-b7f734c956753f38210af45b4651a753

**Comment by GGabriele** [comment]
1) Done. I'll make sure these changes will be propagated to all other PRs.
2) This is done because we want these modules to be supported on both 2.1 and 2.2. @privateip has helped a lot for this, but I don't know of any plan to move this to `module_utils` yet.

**Comment by gregdek** [comment]
Thanks @GGabriele for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by GGabriele** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @GGabriele. To the current maintainers, @jedelman8, @GGabriele, @privateip, @gundalow please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**